### PR TITLE
Add cluster-api-provider-metal3 repo in Prow and Jenkins config

### PR DESCRIPTION
**Pull request by maelk**
This adds the CI to the new repository cluster-api-provider-metal3

### CODE REVIEW

File: jenkins/README.md [hunk 1]
```diff
@@ -26,11 +26,11 @@ publicly visible will get admin rights on the CI jobs. This means :
 
 We have multiple jobs that run some integration tests. The jobs can be
 triggered on PR from metal3-dev-env, baremetal-operator and
-cluster-api-provider-baremetal repositories by commenting the commands below.
+cluster-api-provider-metal3 repositories by commenting the commands below.
 The job result will be posted as a comment.
 
- * **/test-integration** run integration tests for V1alpha1 on Ubuntu
- * **/test-centos-integration** run integration tests for V1alpha1 on CentOS
+ * **/test-integration** run integration tests for master on Ubuntu
```
Comment (fmuyassarov) [inline jenkins/README.md]: shouldn't it be "run v1alpha3 integration tests for master on *..."

Comment (maelk) [reply jenkins/README.md]: As we discussed offline, we'll change the jobs after the release on 10 th of March, so this will be correct then

**Comment by maelk** [comment]
/hold cancel
Just needs lgtm to go in.
/cc @fmuyassarov
/cc @kashifest

**Pull request merged**